lib/awesome_spawn.rb

```diff
@@ -118,7 +119,13 @@ def build_command_line(command, params = nil)
   private
 
   def launch(env, command, spawn_options)
-    output, error, status = Open3.capture3(env, command, spawn_options)
+    capture2e = spawn_options.delete(:capture2e)
```

Comment:
I think this should happen outside of this method...probably in `parse_command_options`, which is where we break the original options into pieces.

Reply:
I feel like in doing so, I would have to then fix method definitions, because it can't stay in `spawn_options` without causing an error.

Reply:
yeah, but all of the definitions are private (it's just launch I think)

Reply:
No, you would basically need to do this (if I am understanding you correctly):

```diff
diff --git a/lib/awesome_spawn.rb b/lib/awesome_spawn.rb
index ab2bc54..5ab432c 100644
--- a/lib/awesome_spawn.rb
+++ b/lib/awesome_spawn.rb
@@ -71,7 +71,7 @@ module AwesomeSpawn
   def run(command, options = {})
     bad_keys = (options.keys.flatten & [:in, :out, :err]).map { |k| ":#{k}" }
     raise ArgumentError, "options cannot contain #{bad_keys.join(", ")}" if bad_keys.any?
-    env, command_line, options = parse_command_options(command, options)
+    env, command_line, options, capture2e = parse_command_options(command, options)
 
     if (in_data = options.delete(:in_data))
       options[:stdin_data] = in_data
@@ -118,8 +118,7 @@ module AwesomeSpawn
 
   private
 
-  def launch(env, command, spawn_options)
-    capture2e = spawn_options.delete(:capture2e)
+  def launch(env, command, spawn_options, capture2e = false)
     if capture2e
       output, status = Open3.capture2e(env, command, spawn_options)
       error          = output
@@ -130,10 +129,11 @@ module AwesomeSpawn
   end
 
   def parse_command_options(command, options)
-    options = options.dup
-    params  = options.delete(:params)
+    options   = options.dup
+    params    = options.delete(:params)
+    capture2e = options.delete(:capture2e)
     env = options.delete(:env) || {}
 
-    [env, build_command_line(command, params), options]
+    [env, build_command_line(command, params), options, capture2e]
   end
 end
```

basically trading one line for about 5 others... and I feel like `parse_command_options` really is for setting `:params` and `:env` anyway, and doesn't really deal with the `launch` part of things.